src/realm/object-store/sync/sync_session.cpp

```diff
@@ -500,21 +502,20 @@ void SyncSession::handle_error(SyncError error)
     else if (error_code.category() == realm::sync::protocol_error_category()) {
         SimplifiedProtocolError simplified =
             get_simplified_error(static_cast<sync::ProtocolError>(error_code.value()));
-        if (error.server_requests_client_reset) {
+        if (error.server_requests_action) {
             // we have specific instructions from the server concerning client resets
-            switch (*error.server_requests_client_reset) {
-                case SyncError::ClientResetModeAllowed::DoNotClientReset:
+            switch (*error.server_requests_action) {
+                case sync::ProtocolErrorInfo::Action::ClientReset:
+                    [[fallthrough]];
+                case sync::ProtocolErrorInfo::Action::ClientResetNoRecovery:
+                    simplified = SimplifiedProtocolError::ClientResetRequested;
+                    break;
+                default:
                     if (simplified == SimplifiedProtocolError::ClientResetRequested) {
```

Comment:
how does this happen?

Reply:
If `ProtocolError::write_not_allowed` is received, the client will attempt a reset. I now checked and saw that the server also sends `ClientResetNoRecovery` so I'll look into simplifying `get_simplified_error()`.